shared/src/main/scala/com/raquo/domtypes/generic/defs/eventProps/FormEventProps.scala

```diff
@@ -49,6 +48,17 @@ trait FormEventProps[
     */
   lazy val onSelect: EP[DomHtmlElementTargetEvent] = eventProp("select")
 
+  /**
+    * The DOM beforeinput event fires when the value of an <input>, or <textarea>
+    * element is about to be modified. The event also applies to elements with
+    * contenteditable enabled, and to any element when designMode is turned on.
+    *
+    * MDN
+    *
+    * @note IE does not support this event.
+    */
+  lazy val onBeforeInput: EP[DomInputEvent] = eventProp("beforeinput")
```

Comment:
I wasn't sure how you like to style these mixed case here. This seemed sensible.